packages/internal-test-env/index.ts

```diff
@@ -0,0 +1,351 @@
+    "string"
+  ) {
+    throw new Error(
+      "The environment variable E2E_TEST_NOTIFICATION_GATEWAY is undefined."
+    );
+  }
+}
+
+export function getNodeTestingEnvironment(): TestingEnvironmentNode {
+  setupEnv();
+  getTestingEnvironment(process.env);
+
+  if (
+    process.env.E2E_TEST_REQUESTOR_CLIENT_ID !== undefined ||
+    process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== undefined
+  ) {
+    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_ID !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_REQUESTOR_CLIENT_ID is undefined."
+      );
+    }
+
+    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_REQUESTOR_CLIENT_SECRET is undefined."
+      );
+    }
+
+    if (typeof process.env.E2E_TEST_VC_PROVIDER !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_VC_PROVIDER is undefined."
+      );
+    }
+  }
+  if (
+    process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_ID !== undefined ||
+    process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_SECRET !== undefined
+  ) {
+    if (typeof process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_ID !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_RESOURCE_OWNER_CLIENT_ID is undefined."
+      );
+    }
+    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== "string") {
+      throw new Error(
+        "The environment variable E2E_TEST_REQUESTOR_CLIENT_SECRET is undefined."
+      );
+    }
+  }
+
+  return {
+    idp: process.env.E2E_TEST_IDP,
+    environment: process.env.E2E_TEST_ENVIRONMENT,
+    protocol: process.env.E2E_TEST_NOTIFICATION_PROTOCOL,
+    notificationGateway: process.env.E2E_TEST_NOTIFICATION_GATEWAY,
+    clientCredentials: {
+      requestor: {
+        id: process.env.E2E_TEST_REQUESTOR_CLIENT_ID,
+        secret: process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET,
+      },
+      resourceOwner: {
+        id: process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_ID,
+        secret: process.env.E2E_TEST_RESOURCE_OWNER_CLIENT_SECRET,
+      },
+    },
+    vcProvider: process.env.E2E_TEST_VC_PROVIDER,
+    features: featuredFlags,
+  };
+}
+
+export function getBrowserTestingEnvironment(): TestingEnvironmentBrowser {
+  setupEnv();
+  getTestingEnvironment(process.env);
+
+  if (process.env.E2E_TEST_USER === undefined) {
+    throw new Error("The environment variable E2E_TEST_USER is undefined.");
+  }
+  if (process.env.E2E_TEST_PASSWORD === undefined) {
+    throw new Error("The environment variable E2E_TEST_PASSWORD is undefined.");
+  }
+
+  return {
+    clientCredentials: {
+      resourceOwner: {
+        login: process.env.E2E_TEST_USER,
+        password: process.env.E2E_TEST_PASSWORD,
+      },
+    },
+    idp: process.env.E2E_TEST_IDP,
+    notificationGateway: process.env.E2E_TEST_NOTIFICATION_GATEWAY,
+    features: featuredFlags,
+  };
+}
+
+export async function getAuthenticatedSession(
```

Comment:
We may wanna rename this to `getOwnerSession`? Or make it generic:

```getAuthenticatedSession("resourceOwner", environment)```

Reply:
well if we are going to rename it to "getOwnerSession" then I would rather call it getResourceOwnerSession cause thats more clear I think, we could also do the second with validation but I dont really think thats necessary cause that means we need to validate that string right? to make sure we have the clientCredentials for resourceOwner?

Reply:
Yeah, `getResourceOwnerSession` also work, or we make it accept a credentials set?